commons-library/src/main/java/com/afh/skytouch/commons/configuration/ProductTransferConfiguration.java

```diff
@@ -0,0 +1,112 @@
+package com.afh.skytouch.commons.configuration;
+
+
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+import org.springframework.amqp.core.Binding;
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.DirectExchange;
+import org.springframework.amqp.core.Exchange;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.annotation.EnableRabbit;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.converter.MappingJackson2MessageConverter;
+
+
+@Configuration
+@EnableRabbit
+public class ProductTransferConfiguration {
+
+    public static final String exchange = "product";
+    public static final String create = "product.create";
+    public static  final String getAll = "product.getAll";
+
+    @Autowired
+    private ConnectionFactory rabbitConnectionFactory;
+
+    @Bean
+    public Exchange productExchange(){
+        return new DirectExchange(exchange);
+    }
+
+    @Bean
+    public Jackson2JsonMessageConverter producerConverter() {
+        Jackson2JsonMessageConverter producerConverter = new Jackson2JsonMessageConverter(mapper());
+        return producerConverter;
+    }
+
+    @Bean
+    public MappingJackson2MessageConverter consumerConverter(){
+        MappingJackson2MessageConverter consumerConverter = new MappingJackson2MessageConverter();
+        consumerConverter.setObjectMapper(mapper());
+        return consumerConverter;
+    }
+
+    @Bean
+    public Queue createQueue(){
```

Comment:
You could use this constructor instead:

public Queue(String name) {
        this(name, true, false, false);
}